v2/util/ca.go

```diff
@@ -44,6 +44,18 @@ func IsSubscriberCert(c *x509.Certificate) bool {
 	return !IsCACert(c) && !IsSelfSigned(c)
 }
 
+func IsOCSPCert(cert *x509.Certificate) bool {
+	if len(cert.ExtKeyUsage) == 0 {
+		return true
+	}
+	for _, eku := range cert.ExtKeyUsage {
+		if eku == x509.ExtKeyUsageAny || eku == x509.ExtKeyUsageOcspSigning {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I'm curious about this as well. Are there other lints you are hoping to exclude from running in their `CheckApplies` if the cert has an OCSP EKU?

It seems somewhat duplicative and maybe we should consider picking one or the other (EKU/KU pulled from a `Has*` function OR `Is*Cert` boolean functions) and not have both methods of collecting the same information. I suspect the Is functions were born out of concern that contributors/integrators might not know all the possible EKU/KU values that could potentially exist on a certificate so we "blessed" a few commonly used ones, but that knowledge could be solved with documentation.

Reply:
> I suspect the Is functions were born out of concern that contributors/integrators might not know all the possible EKU/KU values that could potentially exist on a certificate so we "blessed" a few commonly used ones, but that knowledge could be solved with documentation.

I created the `IsOCSPCert` function to have one central place which decides, whether a certificate is capable of signing OCSP responses or not. I created it analogously to the `IsServerAuthCert` which was already there.

Reply:
@sleevi 

> This contrasts with https://github.com/zmap/zlint/pull/472/files#diff-0e6a1fbcef97b2e7645afdbe65060787 where you just call HasEKU

I changed the method `Initialize` in https://github.com/RufusJWB/zlint/commit/66af7e6beebb4f185efaa28644294f3db8818e74 to use the helper method from the https://github.com/RufusJWB/zlint/blob/master/v2/util/ca.go file. Now we only have one place to clarify, what makes a certificate an OCSP responder certificate

Reply:
Do we all agree on the definition of what makes a certificate an OCSP responder certificate as it is implemented in the lines from above?

Reply:
No, I don’t think that’s correct, from spec or implementation 😅 That’s part of why I was trying to figure out the “why” with my [earlier comment](https://github.com/zmap/zlint/pull/472#discussion_r490533146), because I wasn’t sure if I was missing some context.

Specifically, I don’t think omitting EKU or having AnyEKU makes it an OCSP responder. From memory, a number of libraries require the explicit EKU, although I don’t recall from 6960 if that’s explicitly stated. It certainly “feels” wrong, but at least that makes it easier to sort through now that it’s clear the previous difference (between Has and Is) wasn’t an intentional decision.

Reply:
In [RFC6960 4.2.2.2](https://tools.ietf.org/html/rfc6960#section-4.2.2.2) it is stated:

> Therefore, a certificate's issuer MUST do one of the following:
> 
>    - sign the OCSP responses itself, or
> 
>    - explicitly designate this authority to another entity
> 
>    OCSP signing delegation SHALL be designated by the inclusion of
>    id-kp-OCSPSigning in an extended key usage certificate extension
>    included in the OCSP response signer's certificate.  This certificate
>    MUST be issued directly by the CA that is identified in the request.

So it seems to be clear: A certificate is a delegated OCSP responder certificate, if and only if it includes the id-kp-OCSPSigning extension.

But what about the case, when a CA signs the OCSP response itself? Would this CA certificate be considered as an OCSP responder certificate too? For the purpose of this lint, it is not necessary to decide this question, as the requirement to be tested only applies to delegated OCSP responders. --> I'll rename the method to IsDelegatedOCSPResponderCert .

Reply:
Can we set this conversation on resolved?